roslibrust_codegen/src/lib.rs

```diff
@@ -198,6 +230,26 @@ fn derive_attrs() -> Vec<syn::Attribute> {
     ]
 }
 
+/// Generates the service for a given service file
+/// The service definition defines a struct representing the service an an implementation
+/// of the RosServiceType trait for that struct
+fn generate_service(service: ServiceFile) -> TokenStream {
+    let service_type_name = format!("{}/{}", &service.package, &service.name);
+    let struct_name = format_ident!("{}", service.name);
+    let request_name = format_ident!("{}", service.request_type.name);
+    let response_name = format_ident!("{}", service.response_type.name);
+    quote! {
+        pub struct #struct_name {
+
```

Comment:
I suppose services can't have constants? Or perhaps this is a sufficiently small issue if they can given that someone can get it from the message type that's generated.

Reply:
Constants within a service will end up existing on either their request or response type, no constants live at the "service scope"